megamek/src/megamek/client/ui/swing/MiniReportDisplay.java

```diff
@@ -145,6 +226,39 @@ public MiniReportDisplay(IClientGUI clientgui) {
         doLayout();
     }
 
+    private void filterReport(String k) {
```

Comment:
This function is doing two different things based on a toggle, it should instead be calling two different functions depending on a toggle.